ludwig/models/llm.py

```diff
@@ -89,17 +90,32 @@ def __init__(
         self._random_seed = random_seed
 
         self.model_name = self.config_obj.base_model
+        self.model_config = AutoConfig.from_pretrained(self.config_obj.base_model)
+
+        backend = self.config_obj.backend
+        self.backend_type = backend.type if hasattr(backend, "type") else "local"
```

Comment:
backend is a dictionary coming in, it looks like– so this doesn't work

Reply:
Intriguing, I feel like that's weird because it should just be the config object and not a dictionary